### PR TITLE
fix create symlink and unzipping sources

### PR DESCRIPTION
**Pull request by ubozov**

### CODE REVIEW

File: dapp/dapp_unix.go [hunk 1]
```diff
@@ -30,8 +30,13 @@ func (d *Dapp) controllerHash() string {
 }
 func (d *Dapp) createSymlink() {
 	target := filepath.Join(d.Path, d.Gui.EntryPoint)
+	linkName := path.Base(d.Gui.EntryPoint)
 	link := filepath.Join(util.DesktopPath(),
-		fmt.Sprintf("%s %s", path.Base(d.Gui.EntryPoint), d.Role))
+		fmt.Sprintf(`%s %s`, linkName, d.Role))
```
Comment (ababo) [inline dapp/dapp_unix.go]: Why raw literal here?

File: util/misc.go [hunk 2]
```diff
@@ -197,59 +196,6 @@ func ParseVersion(s string) int64 {
 	return result
 }
 
-// Unzip will decompress a zip archive, moving all files and folders
```
Comment (ababo) [inline util/misc.go]: So pleased when code is removed rather than added :)

File: util/misc_windows.go [hunk 3]
```diff
@@ -105,3 +108,48 @@ func DesktopPath() string {
 	u, _ := user.Current()
 	return filepath.Join(u.HomeDir, "Desktop")
 }
+
```
Comment (ababo) [inline util/misc_windows.go]: Oh, no, it was just moved...

File: util/misc_windows.go [hunk 4]
```diff
@@ -105,3 +108,48 @@ func DesktopPath() string {
...
+	for _, f := range r.File {
+		rc, err := f.Open()
+		if err != nil {
+			return err
+		}
+		defer rc.Close()
+		fpath := filepath.Join(dest, f.Name)
+
+		if f.FileInfo().IsDir() {
+			os.MkdirAll(fpath, f.Mode())
```
Comment (ababo) [inline util/misc_windows.go]: Error handling?

File: dapp/dapp_unix.go (hunk 1 above)
Comment (ubozov) [reply dapp/dapp_unix.go]: updated

File: util/misc_windows.go (hunk 4 above)
Comment (ubozov) [reply util/misc_windows.go]: updated

**Pull request merged**